File: 2020/test/day_03_test.exs
defmodule Day02Test do
  use ExUnit.Case

  import Day03
  import TestHelper

  describe "Day03" do
    test "part1" do
      assert part1(input("day_03_example_1")) == 7
      assert part1(input("day_03")) == 278
    end

    test "part2" do
      assert part2(input("day_03_example_1")) == 336
      assert part2(input("day_03")) == 9_709_761_600
    end
  end
end

File: 2024/day03/day03_test.go
package day03

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 161, Part1("input_example_1"))
	assert.Equal(t, 182619815, Part1("input"))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, 48, Part2("input_example_2"))
	assert.Equal(t, 80747545, Part2("input"))
}
